examples/interaction.rs

```diff
@@ -0,0 +1,292 @@
+}
+
+pub type CounterSize = u32;
+
+pub type Counter = BaseActor<CounterState>;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct CounterState {
+    pub addr: Id,
+    pub counter: CounterSize,
+}
+
+impl Actor for Counter {
+    type Msg = Msg;
+    type State = CounterState;
+    type Timer = InputTimer;
+
+    fn on_start(&self, _id: Id, _o: &mut Out<Self>) -> Self::State {
+        self.initial_state
+    }
+
+    fn on_msg(
+        &self,
+        _id: Id,
+        state: &mut Cow<Self::State>,
+        src: Id,
+        msg: Self::Msg,
+        o: &mut Out<Self>,
+    ) {
+        let (new_state, output_msgs) = match msg {
+            Self::Msg::SupervisorIncrementRequest(n) => (
+                CounterState {
+                    counter: state.counter + n,
+                    ..**state
+                },
+                Vec::new(),
+            ),
+
+            Self::Msg::SupervisorReportRequest() => (
+                **state,
+                vec![(src, Self::Msg::CounterReplyCount(state.counter))],
+            ),
+
+            _ => (**state, Vec::new()),
+        };
+
+        *state.to_mut() = new_state;
+
+        for (addr, msg) in output_msgs {
+            o.send(addr, msg)
+        }
+    }
+}
+
+// Note: The counter is separated into two machines to test message based interaction without any timers.
+pub type Supervisor = BaseActor<SupervisorState>;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct SupervisorState {
+    pub addr: Id,
+    pub threshold: CounterSize,
+    pub counter_addr: Id,
+    pub input_addr: Id,
+    pub success: bool,
+}
+
+impl Actor for Supervisor {
+    type Msg = Msg;
+    type State = SupervisorState;
+    type Timer = InputTimer;
+
+    fn on_start(&self, _id: Id, _o: &mut Out<Self>) -> Self::State {
+        self.initial_state
+    }
+
+    fn on_msg(
+        &self,
+        _id: Id,
+        state: &mut Cow<Self::State>,
+        src: Id,
+        msg: Self::Msg,
+        o: &mut Out<Self>,
+    ) {
+        // InputActor triggers message forwarding.
+        if src == state.input_addr {
+            o.send(state.counter_addr, msg)
+        } else {
+            let (new_state, output_msgs) = match msg {
+                Self::Msg::CounterReplyCount(n) => {
+                    if n >= state.threshold {
+                        (
+                            SupervisorState {
+                                success: true,
+                                ..**state
+                            },
+                            Vec::new(),
+                        )
+                    } else {
+                        (**state, Vec::new())
+                    }
+                }
+
+                _ => (**state, Vec::new()),
+            };
+
+            *state.to_mut() = new_state;
+
+            for (addr, msg) in output_msgs {
+                o.send(addr, msg)
+            }
+        }
+    }
+}
+
+pub struct ExternalInput {
```

Comment:
I interpret "external input" as the messages the actors receive. A more intuitive name for this actor might simply be `Client` or `TestClient`.